### DISCUSSION TITLE
Is Angular 17 SSR officially supported in Azure Static Web Apps?

### DISCUSSION THREAD

**Discussion by anirugu** [Azure/static-web-apps#1362] [Q&A] [answered]
**Describe the bug**

**To Reproduce**

**Expected behavior**

**Screenshots**

**Device info (if applicable):**

**Additional context**

**Comment by thomasgauvin** [comment, marked as answer]
Hi @anirugu, we're aware of this new functionality from Angular but it requires an Express server to host it. As a result, it is not supported on Static Web Apps as of now.

**Reply by iemanuel** [reply]
any news regarding support for angular ssr?

**Reply by quedicesebas** [reply]
Waiting...

**Reply by quedicesebas** [reply]
Should I use Azure functions?

**Reply by thomasgauvin** [reply]
I'd point you into the direction of Analog.js which may have built out adapters support and deployment to Static Web Apps. You would probably need to do some custom hosting to deploy Angular SSR on Static Web Apps.

**Reply by lluthus** [reply]
will angular ssr supported by static web app ?

**Reply by blogcraft** [reply]
and how about now?

**Reply by quedicesebas** [reply]
Go to Firebase App Hosting :)

**Reply by blogcraft** [reply]
No thanks, I don’t like firebase, I just figured out how to launch SSR it in Azure AppService.

**Reply by quedicesebas** [reply]
How?

**Reply by blogcraft** [reply]
It's quite simple actually, you just need to create an AppService Linux (haven't tested in Windows) and choose Node 20 as the stack (I didn't test other versions) and in the AppService configuration there is a place to set up a "Startup Command" and set it like `node server/server.mjs`.

And that's it.

**Reply by SamiV42** [reply]
Thanks for posting your solution @blogcraft! It saved me quite a bit of trial and error.  I can confirm that using your solution it is possible to run an Angular SSR app on an Azure Web App.
